cluster_pack/filesystem.py

```diff
@@ -65,6 +68,23 @@ def _rm(self: Any, path: str, recursive: bool = False) -> None:
             os.rmdir(path)
 
 
+def _rename(self: Any, path: str, new_path: str) -> None:
+    if os.path.exists(new_path):
+        raise OSError(f"File exist at destination {new_path}")
+    os.rename(path, new_path)
+
+
+def _s3_rename(self: Any, path: str, new_path: str) -> None:
+    if self.exists(new_path):
+        raise OSError(f"File exist at destination {new_path}")
+    self.fs.mv(path, new_path)
+
+
+def _hdfs_st_mode(self: Any, path: str) -> int:

```

Comment:
You added this for the tests ? (How could the `put` test have been working before then ? 🤔)

Reply:
- rename was not used
- hdfs_st_mode is not actually used (except in tests I did locally with real hdfs)